plugins/com.google.cloud.tools.eclipse.appengine.libraries.test/src/com/google/cloud/tools/eclipse/appengine/libraries/AppEngineLibrariesInPluginXmlTest.java

```diff
@@ -41,20 +41,30 @@
 public class AppEngineLibrariesInPluginXmlTest {
 
   private static final String APP_ENGINE_API_LIBRARY_ID = "appengine-api";
+  private static final String APP_ENGINE_ENDPOINTS_LIBRARY_ID = "appengine-endpoints";
+  private static final String OBJECTIFY_LIBRARY_ID = "objectify";
 
   @Test
   public void testAppEngineAPIConfig() throws URISyntaxException, LibraryFactoryException {
     IConfigurationElement[] configurationElements =
         RegistryFactory.getRegistry()
           .getConfigurationElementsFor(AppEngineLibraryContainerInitializer.LIBRARIES_EXTENSION_POINT);
     for (IConfigurationElement configurationElement : configurationElements) {
-      if (configurationElement.getAttribute("id").equals(APP_ENGINE_API_LIBRARY_ID)) {
-        Library appEngineLibrary = new LibraryFactory().create(configurationElement);
-        verifyAppEngineLibrary(appEngineLibrary);
-        return;
+      Library library = new LibraryFactory().create(configurationElement);
+      switch (configurationElement.getAttribute("id")) {
```

Comment:
I wanted to assert on the content of the xml as a whole, not partially by looking up each library definition one by one, but I can refactor this into 4 test methods, one to test the cardinality of library definitions in the plugin.xml and one for each of the expected libraries.